File: split/newligature.pas
function newligature(f, c: quarterword; q: halfword): halfword;
var
  p: halfword;
begin
  p := getnode(2);
  mem[p].hh.b0 := 6;
  mem[p + 1].hh.b0 := f;
  mem[p + 1].hh.b1 := c;
  mem[p + 1].hh.rh := q;
  mem[p].hh.b1 := 0;
  newligature := p;
end;

File: cpp/newligature.cpp
#include "newligature.h"
#include "getnode.h"

halfword newligature(quarterword f, quarterword c, halfword q)
{
	auto p = getnode(2);
	type(p) = ligature_node;
	type(p+1) = f;
	subtype(p+1) = c;
	link(p+1) = q;
	subtype(p) = 0;
	return p;
}
